app/controllers/application_controller.rb

```diff
@@ -73,31 +73,28 @@ def find_platform(param = :id)
   end
 
   def find_project
-    @project = Project.find_with_includes!(params[:platform], params[:name], [:repository, :versions])
+    @project = Project.find_exact!(params[:platform], params[:name], [:repository, :versions])
     @color = @project.color
   rescue ActiveRecord::RecordNotFound
-    raise if params[:name].blank? || params[:platform]&.downcase != "go"
-
-    resolved_name = PackageManager::Go.resolved_name(params[:name])
-    if resolved_name != params[:name] && Project.known?(params[:platform], resolved_name)
-      redirect_to(
-        # Unescape since url_for automatically escapes our already-escaped resolved_name
-        URI.unescape(
-          url_for(
-            params
-              .to_unsafe_h
-              .merge({ "name" => CGI.escape(resolved_name) })
-          )
+    raise if params[:name].blank?
+
+    project = Project.find_fuzzy!(params[:platform], params[:name])
+    raise if project.name == params[:name]
```

Comment:
I know this was how it was originally written, but I think we could use `Project.find_best` in here and if the name does not equal `params[:name]` do the redirect. I think would make this look simpler.

Reply:
Good thinking, lemme look at that